.github/workflows/install-valkey/action.yml

```diff
@@ -19,30 +19,52 @@ inputs:
 
 env:
     CARGO_TERM_COLOR: always
-    VALKEY_MIN_VERSION: "7.2.5"
+    VALKEY_MIN_VERSION: "7.2"
 
 runs:
     using: "composite"
 
     steps:
         - name:
               Cache Valkey
-              # TODO: remove the musl ARM64 limitation when https://github.com/actions/runner/issues/801 is resolved
-          if: ${{ inputs.target != 'aarch64-unknown-linux-musl' }}
+              # TODO: self-hosted runners are actually cloning the repo, using the cache from the previous run
+              # will not work as expected. We need to find a way to cache the valkey repo on the runner itself.
+          if: ${{!contains(inputs.target, 'aarch64') }}
           uses: actions/cache@v4
           id: cache-valkey
           with:
               path: |
                   ~/valkey
               key: valkey-${{ inputs.engine-version }}-${{ inputs.target }}
 
+        - name: Build Valkey for ARM
+          if: ${{ contains(inputs.target, 'aarch64') }}
```

Comment:
I think this script is applicable for any target and runner, so you can remove the second, but fix `if` condition there

Reply:
I'll check it

Reply:
Base on previous comment, they are pretty differnt in the logic, the one clean and rebuild, the other or already have it or not, if has, and check it based on cache.